packages/compass-connections/src/stores/connections-store.spec.tsx

```diff
@@ -436,6 +436,41 @@ describe('useConnections', function () {
       ).to.exist;
     });
 
+    it('should create a name if there is none', async function () {
+      const connectionWithoutName: ConnectionInfo = {
+        id: '123',
+        connectionOptions: {
+          connectionString: 'mongodb://localhost:27017',
+        },
+        favorite: {
+          name: '',
+          color: 'color2',
+        },
+        savedConnectionType: 'recent',
+      };
+      mockConnectionStorage = new InMemoryConnectionStorage([
+        connectionWithoutName,
+      ]);
+      const connections = renderHookWithContext();
+
+      // Waiting for connections to load first
+      await waitFor(() => {
+        expect(connections.current.recentConnections).to.have.lengthOf.gt(0);
+      });
+
+      await connections.current.duplicateConnection(connectionWithoutName.id, {
+        autoDuplicate: true,
+      });
+
+      await waitFor(() => {
+        expect(
+          connections.current.recentConnections.find((info) => {
+            return info.favorite.name === 'localhost:27017 (1)';
```

Comment:
Related to my other comment, let's extend the test to make sure we're not loosing the color